Peer.Connectors/GithubPullRequestSource.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Octokit;
+using Peer.Domain;
+using Peer.Domain.Models;
+using Peer.User;
+
+namespace Peer.Connectors
+{
+    public class GithubPullRequestSource : IPullRequestSource
+    {
+        private static readonly Regex _paramRegex = new Regex(@"(?<orgName>[\w\-]+)/(?<repoName>[\w\-]+)/issues/\d+");
+        private readonly ConfigModel _config;
+        private static IClient<GithubClient> _user;
+
+        public GithubPullRequestSource(ConfigModel config, IClient<GithubClient> user)
+        {
+            _config = config;
+            _user = user;
+        }
+
+        public async Task<IEnumerable<PeerPullRequest>> FetchPullRequests()
+        {
+            var userclient = _user.CreateClient().GitClient;
```

Comment:
Are you intending to extend IClient to do something other than just create the client? Otherwise you can probably just inject the client or an IEnumerable of them and save the type since it doesn't seem to be doing a ton other than wrapping

Reply:
It will be expanded to handle the client's lifecycle. Aiming for other methods like "destroy"

Reply:
Can probably just use a non-punned GitHubClient that implements IDisposable if you need to handle the lifecycle specifically. Just doesn't feel like the IClient itself really adds a ton to the interactions but I might be missing something